### ISSUE TITLE
400 Client Error: BAD REQUEST

### ISSUE THREAD

**Issue by yudhiahmadi** [luxonis/blobconverter#34]
I am converting an OpenVino IR model and I get the following error:

HTTPError: 400 Client Error: BAD REQUEST for url: https://blobconverter.luxonis.com/compile?version=2021.4&no_cache=False

How should i go about debugging this issue?

Thanks

**Comment by HassanAlmlhem** [comment]
I have the same error, were you able to solve it?

**Comment by VanDavv** [comment]
@yudhiahmadi @HassanAlmlhem together with the error code, you should see the following response body as well (added breaks for better formatting)

```
{ 
"exit_code": 1, 
"message": "Command failed with exit code 1, command: /opt/intel/openvino/deployment_tools/inference_engine/lib/intel64/myriad_compile -m /tmp/blobconverter/c7d2ecf91d79439fbc8597d42516336d/mde/FP16/mde.xml -o /tmp/blobconverter/c7d2ecf91d79439fbc8597d42516336d/mde/FP16/mde.blob -c /tmp/blobconverter/c7d2ecf91d79439fbc8597d42516336d/myriad_compile_config.txt -ip U8", 
"stderr": "Unsupported pad value\n", 
"stdout": "Inference Engine: \n\tIE version ......... 2021.4.0\n\tBuild ........... 2021.4.0-3839-cd81789d294-releases/2021/4\n\u001b[1;33m[Warning][VPU][Config] Deprecated option was used : VPU_MYRIAD_PLATFORM\u001b[0m\n" 
}
```

This should give you insights on what went wrong with the conversion

**Comment by Erol444** [comment]
@ariyapour could you share xml/bin or pt/onnx you are trying to convert/compile?

**Comment by ariyapour** [comment]
Sure. I downloaded the yolov8n.pt from this [link](https://github.com/ultralytics/ultralytics#models). It sould be a default pretrained model. I converted it to Onnx and then tried to convert it to blob using the blobconverter:
`blobconverter --onnx-model ./yolov8n.onnx --shaves 6`
and the output is:
![error](https://user-images.githubusercontent.com/7849979/234122266-13afcba9-b3f2-4b9c-b88c-75afb66c68bf.png)

then I tried to convert the same onnx model on your website in this [linke](https://blobconverter.luxonis.com/) and the conversion was successfull.
I can see the error messages and the fact it is complaining about the model, but I am wondering why it is not working on my local machine while it works online. I appreciate your help.

Link to the onnx model I used for this experiment: 
[yolov8n_onnx.zip](https://github.com/luxonis/blobconverter/files/11314798/yolov8n.zip)

**Comment by tersekmatija** [comment]
Hey @ariyapour,

Thanks for reporting this. The issue is that `blobconverter` CLI actually uses OpenVINO 2021.4 by default. If you try to convert the same ONNX through the website and choose OpenVINO 2021.4 you'll see that it throws an issue regarding the shapes.
```[ ERROR ]  Cannot infer shapes or values for node "/model.22/dfl/Reshape".
[ ERROR ]  Number of elements in input [   1   72 8400] and output [1, 4, 16, 8400] of reshape node /model.22/dfl/Reshape mismatch
[ ERROR ]  
[ ERROR ]  It can happen due to bug in custom shape infer function <function Reshape.infer at 0x7f8fb26cb700>.
[ ERROR ]  Or because the node inputs have incorrect values/shapes.
[ ERROR ]  Or because input shapes are incorrect (embedded to the model or passed via --input_shape).
[ ERROR ]  Run Model Optimizer with --log_level=DEBUG for more information.
[ ERROR ]  Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.middle.PartialInfer.PartialInfer'>): Stopped shape/value propagation at "/model.22/dfl/Reshape" node. 
```
This is fixed in OpenVINO 2022.1, which is set as default on the website, and thus the conversion does not fail. If you still want to use the CLI, make sure to include `--version 2022.1` flag when executing.

However, if you want to use Yolo with our YoloDetectionNetwork node, I'd suggest you to use https://tools.luxonis.com/ instead. You can then check out [this example](https://github.com/luxonis/depthai-experiments/tree/master/gen2-yolo/device-decoding) to see how it can be used with our device.
